src/main/java/org/primefaces/component/calendar/BaseCalendarRenderer.java

```diff
@@ -163,20 +174,49 @@ public Object getConvertedValue(FacesContext context, UIComponent component, Obj
             throw e;
         }
 
-        //Use built-in converter
-        format = new SimpleDateFormat(uicalendar.calculatePattern(), uicalendar.calculateLocale(context));
-        format.setLenient(false);
-        format.setTimeZone(CalendarUtils.calculateTimeZone(uicalendar.getTimeZone()));
         try {
-            return format.parse(submittedValue);
+            //TODO: is there some way to dynamically determine the type?
```

Comment:
which type?

Reply:
The type of the value which is bound to the component.
This would allow us to do something like the following pseudocode:
```
if (typeOfValue == java.util.Date) {
   //convert to java.util.Date
}
else if (typeOfValue == LocalDate)
{
   //convert to java.time.LocalDate
}
else if (typeOfValue == java.time.LocalDateTime)
{
   //convert to java.time.LocalDateTime
}
else if (typeOfValue == java.time.ZonedDateTime)
{
   //convert to java.time.ZonedDateTime
}
```
So we won´t need a converter for java.util.Date, we would be able to support ZonedDateTime.

Reply:
Maybe i simply have to look a few lines up.
```
ValueExpression ve = uicalendar.getValueExpression("value");
            if (ve != null) {
                Class type = ve.getType(context.getELContext());
```
I will try this.